data-semantics.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: page
+title: Data Semantics Guidelines
+---
+<div id="toc"></div>
+
+## Introduction
+
+OpenTracing defines an API through which application instrumentation can log data to a pluggable tracer. In general, there is no guarantee made by OpenTracing about the way that data will be handled by an underlying tracer. That said, the APIs are by design open-ended: what type of data should be provided to them?
+
+A general understanding between the instrumentor and tracer developers adds great value: if certain known tag key/values are used for common application scenarios, tracers can choose to pay special attention to them. The same is true of `log`ged events, and span structure in general.
+
+As an example, consider the common case of a HTTP-based application server. The URL of an incoming request that the application is handling is often useful for diagnostics, as well as the HTTP verb and the resultant status code. An instrumentor could choose to report the URL in a tag named `URL`, or perhaps named `http.url`--either would be valid from the pure API perspective. But if the Tracer wishes to add intelligence, such as indexing on the URL value or sampling proactively for requests to a particular endpoint, it must know where to look for relevant data. In short, when tag names and other instrumentor-provided values are used consistently, the tracers on the other side of the API can employ more intelligence.
+
+The guidelines contained describe a common ground on which instrumentors and tracer authors can build beyond pure data collection. Adherence to the guidelines is optional but highly recommended for instrumetors.
+
+
+
+## Spans
+
+
+### Span Naming
+
+Spans carry tags, logs, and attributes, but they also have a top-level **operation name**. This should be a low-cardinality string value representing the type of work being done in the span. Typically, a process, framework, library, or module name is a good guideline.
+
+For instance, a request being handled by a Flask-based application might be instrumented using a WSGI middleware. In this situation, the first span operation name might be `WSGI`. If the application subsequently makes a database query, the span name might be `SQL`.
```

Comment:
I vote for (4). For what it's worth, I would argue strenuously against (1). I would also argue strenuously against (3): it makes sense for monkey-patched instrumentation, but OT is designed for explicit instrumentation and the operation name is going to be more descriptive semantically. (2) is conceivable but feels burdensome... though maybe I'm just too accustomed to the status quo?

I appreciate why `component` can be useful, but (a) information about the package can be captured automatically by the tracer impl, and (b) it is easier to add `setComponentName` later than to take it away. So, yes, my vote is (4) with the option to pursue (2) in the future once we see how useful it is to have both.